src/gpl/src/cudasplibs.cu

```diff
@@ -0,0 +1,119 @@
+#include "cudasplibs.h"
+#include "utl/Logger.h"
+
+void cudasplibs::cudaerror(cudaError_t code) {
+    if (code != cudaSuccess){
+        log_->report("[CUDA ERROR] {} at line {} in file {} \n", cudaGetErrorString(code), __LINE__, __FILE__);
+        cudaDeviceReset();
+    }
+}
+void cudasplibs::cusparseerror(cusparseStatus_t code) {
+    if (code != CUSPARSE_STATUS_SUCCESS){
+        log_->report("[CUSPARSE ERROR] {} at line {} in file {}\n", cusparseGetErrorString(code), __LINE__, __FILE__); 
+        cudaDeviceReset();
+    }
+}
+
+void cudasplibs::cusolvererror(cusolverStatus_t code) {
+    if (code != CUSOLVER_STATUS_SUCCESS){
+        log_->report("[CUSOLVER ERROR] {} at line {} in file {}\n", cudaGetErrorString(*(cudaError_t*)&code), __LINE__, __FILE__); 
+        cudaDeviceReset();
+    }
+}
+
+cudasplibs::cudasplibs(std::vector<int>& cooRowIndex, std::vector<int>& cooColIndex, std::vector<float>& cooVal, Eigen::VectorXf& fixedInstForceVec, utl::Logger* logger){
+    m = fixedInstForceVec.size();
+    nnz = cooVal.size();
+    log_ = logger;
+    // Allocate device memeory and copy data to device
+    cudaerror(cudaMalloc((void**)&d_cooRowIndex, nnz * sizeof(int)));
```

Comment:
Same comment as last time - you should be using RAII classes not low level management.

Reply:
CUDA thrust is being used instead of the Malloc. The vectors allocated by thrust can be released automatically when it is out of scope.